python/ray/tests/test_actor.py

```diff
@@ -1393,6 +1393,27 @@ def get_actor(actor):
     assert origin == remote
 
 
+@pytest.mark.skipif(
+    sys.version_info < (3, 9), reason="GenericAlias is new in Python 3.9"
+)
+def test_classmethod_genericalias():
+    """
+    The Python built-in function `inspect.signature` doesn't support
+    classmethod(GenericAlias). Hence, if we call `inspect.signature(MyClass)`,
+    it will raise a ValueError.
+    """
+    from types import GenericAlias
+
+    @ray.remote
+    class MyClass:
+        __class_getitem__ = classmethod(GenericAlias)
+
+    # This would allow MyClass[int] to behave similarly to specifying a generic type
+    myClass = MyClass.remote()
+    obj_ref = myClass.__class_getitem__.remote(int)
+    print(ray.get(obj_ref))
```

Comment:
I don't think people will do this. I think we can remove these.

Maybe add another example that's also failing:

```
@ray.remote
class MyIterable(Iterable):
   def __iter__(self):
       return iter(range(0, 10))
```

Reply:
I prefer to use the existing one because it directly tests the actual root cause. `MyIterable(Iterable)` has the same issue because it also employs `__class_getitem__ = classmethod(GenericAlias)` ([code](https://github.com/python/cpython/blob/f366e215044e348659df814c27bf70e78907df21/Lib/_collections_abc.py#L294)) under the hood. In my opinion, there is no need to add another layer to hide the actual root cause.

Reply:
I can update it if you have a strong opinion on that matter.

Reply:
Oh yea, keep yours. I'm just wondering do we need 

```
# This would allow MyClass[int] to behave similarly to specifying a generic type
    myClass = MyClass.remote()
    obj_ref = myClass.__class_getitem__.remote(int)
    print(ray.get(obj_ref))
```
since the exception is thrown before it and I don't think this is something people will do.